api_tests/waffle/views/test_waffle_cookies.py

```diff
@@ -0,0 +1,83 @@
+import mock
+import pytest
+import waffle
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    FlagFactory,
+    ProjectFactory
+)
+from website.ember_osf_web.decorators import camouflage_request
+
+from website.project.views.node import _view_project
+from functools import wraps
+from api.base.views import root
+from framework.auth.core import _get_current_user
+from tests.json_api_test_app import JSONAPITestApp
+from tests.base import OsfTestCase
+
+
+class MockUser(object):
+    is_authenticated = False
+
+
+def check_for_test_flag(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        from flask import request
+        request.user = _get_current_user() or MockUser()
+        request = camouflage_request(request)
+        waffle.flag_is_active(request, 'test_flag')
+        return func(self, *args, **kwargs)
+    return wrapped
+
+
+@pytest.mark.django_db
+class TestWaffleCookies(OsfTestCase):
+
+    def setUp(self):
+        super(TestWaffleCookies, self).setUp()
+        self.flag = FlagFactory(name='test_flag')
+        self.flag.percent = 50
+        self.flag.everyone = None
+        self.flag.save()
+
+    @pytest.mark.enable_bookmark_creation
+    @mock.patch('website.project.views.node._view_project', check_for_test_flag(_view_project))
+    def test_waffle_leaves_cookie(self):
+        """
+        Tests that django-waffle cookies work in the with our Flask requests.
+
+        Flask waffle cookies are formatted:
+        `dwf_test_flag=True; Expires=True; Max-Age=2592000; Path=/`
+
+        """
+        node = ProjectFactory(is_public=True)
+        user = AuthUserFactory()
+        resp = self.app.get(node.web_url_for('view_project'), auth=user.auth, auto_follow=True)
+
+        waffle_cookie, csrf_cookie = [value for key, value in resp.headers.items() if key == 'Set-Cookie']
+        print waffle_cookie
+        cookie_str = 'dwf_test_flag={}; Expires=True; Max-Age=2592000; Path=/'
+
+        assert waffle_cookie == cookie_str.format('True') or waffle_cookie == cookie_str.format('False')
+        assert 'api-csrf=' in csrf_cookie
+
+    @pytest.mark.enable_bookmark_creation
+    @mock.patch('api.base.views.root', check_for_test_flag(root))
+    def test_waffle_v2_json_leaves_cookie(self):
+        """
+        Tests that django-waffle cookies work in the with our Flask requests.
+
+        DRF waffle cookies are formatted:
+        `dwf_test_flag=True; expires=Fri, 09-Aug-2019 16:33:52 GMT; Max-Age=2592000; Path=/; secure`
+
+        """
+        app = JSONAPITestApp()
+        resp = app.get('/v2/')
+
+        waffle_cookie = [value for key, value in resp.headers.items() if key == 'Set-Cookie'][0]
```

Comment:
We should consider headers unordered. Don't assume the `Set-Cookie` header you're looking for is first.
```suggestion
        waffle_cookie = next(
            value for key, value in resp.headers.items() if key == 'Set-Cookie' and value.startswith('dwf_test_flag'),
            None,
        )
```